### DISCUSSION TITLE
[General] zkSync contract SystemConfig parameters

### DISCUSSION THREAD

**Discussion by enochSatis** [zkSync-Community-Hub/zksync-developers#89] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Continuing the integration of zkSync official bridge in #79 , I found out that there are plenty of preset constant parameters in `SystemConfig.json` (https://github.com/matter-labs/era-contracts/blob/3a4506522aaef81485d8abb96f5a6394bd2ba69e/SystemConfig.json). 

### What is the specific issue or error you're encountering?

Is there any way to check these parameters on-chain, just in case zkSync will update their value in real time (as suggested by @vladbochok about the parameter for `REQUIRED_L2_GAS_PRICE_PER_PUBDATA` in #79 ) ?

### Can you share the error messages or logs you're receiving, if any?

na

### Have you made any recent changes to the contract before encountering this issue?

na

### Are there any external libraries or contracts that your contract interacts with?

na

### Can you provide the relevant portions of your contract code where the issue is occurring?

na

### Have you tried to isolate the problem, and if so, what were the results?

na

### What steps have you already taken to try to resolve the issue?

na

**Comment by MexicanAce** [comment]
@vladbochok Per your [comment](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/79#discussioncomment-6618434) from #79 :

Perhaps this can be resolved by including fields like `REQUIRED_L2_GAS_PRICE_PER_PUBDATA ` in the [SystemContext.sol](https://github.com/matter-labs/era-system-contracts/blob/main/contracts/SystemContext.sol)?

**Reply by enochSatis** [reply]
It would be for zkSync to add these variables to a contract if these variables will be changed in the future.

@uF4No

**Reply by vladbochok** [reply, marked as answer]
@MexicanAce That would be nice to do, but the mechanism for selecting "fair" l2GasPricePerPubdata is to be developed and it is not obvious that it can be efficiently and trustless stored in the L2 contract. We would consider this option when develop this.

**Reply by MexicanAce** [reply]
Thank you for the additional context @vladbochok . Closing this Discussion as this is a feature request to be considered in future iterations of the network node.

**Comment by bradfordbradv** [comment]
To check the parameters specified in SystemConfig.json on-chain, you would typically need to deploy a contract that reads these parameters from the JSON file and exposes them as contract variables or functions. Then, you can call these contract functions to retrieve the parameters on-chain. Alternatively, you can directly embed these parameters into your contracts during deployment, ensuring that they are always up-to-date with the latest values.

**Comment by bartlettdv** [comment]
Environment: Testnet
Issue: Need to check preset constant parameters in SystemConfig.json on-chain for real-time updates.
Potential Solution: Implement a mechanism to fetch and verify these parameters on-chain using smart contracts or off-chain scripts.

**Comment by lupanncoofonpowell** [comment]
Currently, there isn't a direct on-chain method to check the parameters defined in SystemConfig.json. However, you can create a custom contract or off-chain service that periodically fetches and monitors these parameters from zkSync's official sources or network updates.

**Comment by spatounmas** [comment]
There isn't a direct on-chain method to check the parameters defined in SystemConfig.json of zkSync. These parameters are typically set during the deployment of zkSync contracts and are not meant to be modified frequently. However, you can refer to zkSync's documentation or announcements for any updates or changes to these parameters.

**Comment by dianocsemintrey** [comment]
The user is integrating the zkSync official bridge and has discovered preset constant parameters in SystemConfig.json. The user is wondering if there is a way to check these parameters on-chain in case they are updated in real-time by zkSync. No error messages or logs are provided, and no recent changes have been made to the contract. No external libraries or contracts are mentioned, and no relevant portions of the contract code are provided. No steps have been taken to isolate the problem, and no repo link is provided.
